src/StreamJsonRpc.Tests/JsonRpcWithFatalExceptionsTests.cs

```diff
@@ -170,6 +173,50 @@ public async Task CancelMayStillReturnErrorFromServer()
         Assert.False(this.serverRpc.IsDisposed);
     }
 
+    [Fact]
+    public async Task DisposedClientResultsInCancelled()
+    {
+        using (var cts = new CancellationTokenSource())
+        {
+            var invokeTask = this.clientRpc.InvokeWithCancellationAsync<string>(nameof(Server.AsyncMethodFaultsAfterCancellation), new[] { "a" }, cts.Token);
+            await this.server.ServerMethodReached.WaitAsync(this.TimeoutToken);
+            this.clientRpc.Dispose();
+            this.server.AllowServerMethodToReturn.Set();
+
+            // Connection was closed before error was sent from the server
+            await Assert.ThrowsAnyAsync<OperationCanceledException>(() => invokeTask);
+            Assert.Equal(0, this.serverRpc.IsFatalExceptionCount);
+        }
+
+        Assert.True(((IDisposableObservable)this.clientMessageHandler).IsDisposed);
+        Assert.True(((IDisposableObservable)this.serverMessageHandler).IsDisposed);
```

Comment:
This line caused the test to fail non-deterministically. There's no guarantee at this point that the server has already been disposed of, since it responds to the client's disconnection asynchronously.

Reply:
I removed this line.